### PR TITLE
modify resizeImage function

### PR DESCRIPTION
**Pull request by econdepe**
I added a new parameter. When both `maxWidth` and `maxHeight` are specified, there are two resizes possible (resize so that `width = maxWidth` or `height = maxHeight`). By default, the function chooses the resize that will preserve `width <= maxWidth && height <= maxHeight`. But in some cases we may want to choose the "bigger" resize that will violate one of these conditions.

For the moment, this function is only used in `contacts`, where the PR https://github.com/ProtonMail/proton-contacts/pull/288 will take into account this new behavior.

### CODE REVIEW

File: lib/helpers/image.js [hunk 1]
```diff
@@ -66,7 +73,7 @@ export const resizeImage = (
  * @param {String} str
  * @returns {Promise} {mime, base64}
  */
-const extractBase64Image = (str = '') => {
+export const extractBase64Image = (str = '') => {
```
[the remark this answers is not in the document]
Comment (econdepe) [reply lib/helpers/image.js]: yep, I was doing something different before and forgot to remove the `export`

File: test/helpers/image.spec.js [hunk 2]
```diff
@@ -30,58 +30,67 @@ describe('toFile', () => {
 
 describe('resizeImage', () => {
     it('it should be a string', async () => {
-        const base64str = await resizeImage(img, 100, 100);
+        const base64str = await resizeImage({ original: img, maxWidth: 100, maxHeight: 100 });
 
         expect(typeof base64str).toBe('string');
```
[the remark this answers is not in the document]
Comment (econdepe) [reply test/helpers/image.spec.js]: yeah, I agree the tests are a bit prehistoric right now. But they capture basic behavior and test the edge cases in there

File: lib/helpers/image.js [hunk 3]
```diff
@@ -19,38 +19,45 @@ export const toImage = (base64str) => {
...
     original,
     maxWidth = 0,
     maxHeight = 0,
     finalMimeType = 'image/jpeg',
-    encoderOptions = 1
-) => {
+    encoderOptions = 1,
+    bigResize = false
+}) => {
     return toImage(original).then((image) => {
```
[the remark this answers is not in the document]
Comment (econdepe) [reply lib/helpers/image.js]: Sounds good to me. Gonna unify the code and use async/await everywhere
